src/JSInterop/Microsoft.JSInterop/src/Infrastructure/DotNetDispatcher.cs

```diff
@@ -203,13 +203,16 @@ private static void EndInvokeDotNetAfterTask(Task task, JSRuntime jsRuntime, in
                 var parameterType = parameterTypes[index];
                 if (reader.TokenType == JsonTokenType.StartObject && IsIncorrectDotNetObjectRefUse(parameterType, reader))
                 {
+                    jsRuntime.ResetByteArraysToBeRevived();
```

Comment:
What do you think about if we *didn't* call this before each of the possible failure cases? I know that means some data will hang around for longer than we need, but (1) it's limited to "max signalr message size", so isn't very much, and (2) always gets cleared out before the next incoming call anyway.

Limiting the cleanup to only happen in the "received ID=0" or "finished receiving" cases means there's less for us to remember to do as we evolve this logic in the future, and clarifies that the only thing we actually rely on is that it gets cleared in those cases.

Reply:
Alternatively, why don't we move this to the callsite (`Hub` and equivalent Wasm callsite) and clear it after every JS -> DotNet call completes? (Whether successfully or with a failure)

It's not clear to me what we win by using `0` as a way to reset the buffer. Afaik, we should never be in a situation where the buffer is not empty and we receive a stream with a `0` sequence id. That would mean that the previous JS interop call didn't complete successfully.

Reply:
> That would mean that the previous JS interop call didn't complete successfully.

Yes, this was my suggestion for how we can be certain to auto-clean-up in that situation. I agree that a valid alternative is `try`/`finally` around anything that could have failed the previous time to guarantee it was cleaned up already.

TBH I don't mind either way but it's nice to know that even if we make a mistake with forgetting to clean up somewhere, e.g., after some future refactoring when this point gets forgotten about, things will remain safe and errors will only refer to the original cause rather than knock-on effects like breaking *future* JS interop calls.

Reply:
The only case I can think this makes sense is if/when the connection is lost in the middle of a JS interop call (since now it spans multiple messages). I think based on that, it's fine to have it here, I was just a bit worried that we might introduce a hard to find bug in this area where we unknowingly reset the list by mistake. It might be better to be "explicit" about this by forcing the client to send a "reset" frame (with a -1 id) after a failed JS interop call.

In either case, I don't mind this too much, so I think what we have here is good enough.

Reply:
> I think what we have here is good enough.

+1